extensions/notebook/src/protocol/notebookUriHandler.ts

```diff
@@ -0,0 +1,131 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import * as azdata from 'azdata';
+
+import * as request from 'request';
+import * as path from 'path';
+import * as querystring from 'querystring';
+import * as nls from 'vscode-nls';
+const localize = nls.loadMessageBundle();
+import { IQuestion, QuestionTypes } from '../prompts/question';
+import CodeAdapter from '../prompts/adapter';
+import { getErrorMessage, isEditorTitleFree } from '../common/utils';
+
+export class NotebookUriHandler implements vscode.UriHandler {
+	private prompter = new CodeAdapter();
+
+	constructor() {
+	}
```

Comment:
I realize it really doesn't matter in the whole scheme of things but we could always initialize prompter in the constructor

Reply:
I chose this way 😄 I prefer just 1 line for define + create instead of 2. Hope that's OK?